src/components/Button/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { PropsWithChildren, MouseEvent } from 'react';
+import { styButtonPrimary, styUndefinedBtn } from './styles';
+import { ButtonProps } from './buttonProps';
+import { isValidHttpUrl, Route } from './Route';
+import { cx } from '@emotion/css';
+
+const Button = (props: PropsWithChildren<ButtonProps>): JSX.Element => {
+    const { btnSize, btntype, btnColor, link, element, children, ...rest } =
+        props;
+    const className =
+        btnSize !== undefined
+            ? `${btnSize} ${styButtonPrimary}`
+            : cx(styButtonPrimary, styUndefinedBtn);
+    const color = { backgroundColor: btnColor };
+
+    const handleClick = (e: MouseEvent): void => {
+        if (link !== undefined) {
+            if (element !== undefined) {
+                if (e.metaKey || e.ctrlKey) {
+                    return;
+                }
+                e.preventDefault(); // cegah loading
+                Route(link, element);
+            } else isValidHttpUrl(link);
+        }
+    };
+
+    return (
+        <div>
+            <button
+                onClick={handleClick}
+                {...rest}
+                style={color}
+                className={className}
+            >
```

Comment:
```suggestion
            <button
                {...rest}
                onClick={handleClick}
                style={color}
                className={className}
            >
```

in this case order is matter, if you put `{...rest}` here.. if there is onClick attributes inside "rest" it will override the one you have made.. but if you put it on top.. everything below them will overrides everything in top of them.. so make sure that is your case.